compiler/src/main/kotlin/motif/compiler/XFunSpec.kt

```diff
@@ -90,9 +77,10 @@ object XFunSpec {
                 .forEach { funBuilder.addTypeVariable(it) }
     */
 
-    method.parameters.forEach {
+    method.parameters.forEachIndexed { index, parameter ->
       funBuilder.addParameter(
-          ParameterSpec.builder(it.name, it.type.typeName.toKTypeName()).build())
```

Comment:
This `toKTypeName` eventually creates an infinite loop somewhere in kotlin poet due to the unresolved type references itself in the generic type parameter.